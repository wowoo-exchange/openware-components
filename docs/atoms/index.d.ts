export * from './Avatar/Avatar';
export * from './Button/Button';
export * from './Checkbox/Checkbox';
export * from './CryptoIcon/CryptoIcon';
export * from './Input/Input';
export * from './Loader/Loader';
export * from './HeaderItem/HeaderItem';
export * from './Alert/Alert';
export * from './AlertIcon/AlertIcon';
export * from './Pagination/Pagination';
export * from './CloseButton/CloseButton';
export * from './PercentageButton/PercentageButton';
export * from './SummaryField/SummaryField';
