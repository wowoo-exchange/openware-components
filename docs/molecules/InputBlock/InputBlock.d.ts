import * as React from 'react';
interface InputBlockProps {
    /**
     * Additional class name for styling. By default element receives `cr-input-block` class
     * @default empty
     */
    className?: string;
    /**
     * Code of the cryptocurrency
     * @default empty
     */
    currency?: string;
    /**
     * Function for getting value of input tag
     * @default empty
     */
    handleChangeValue: (value: string) => void;
    /**
     * Placeholder for Input component
     * @default empty
     */
    placeholder?: string;
    /**
     * Message for input
     */
    message: string;
    /**
     * Type of Input component
     * @default 'text'
     */
    type?: string;
    /**
     * Value of Input component
     * @default ''
     */
    value: string | number;
    /**
     * Function for handling 'Enter' key
     */
    onKeyPress?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
}
/**
 * Input component with ability to render label with cryptocurrency name.
 */
declare const InputBlock: React.FunctionComponent<InputBlockProps>;
export { InputBlock, InputBlockProps, };
