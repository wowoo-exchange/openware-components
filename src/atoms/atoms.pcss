@import "Avatar/Avatar.pcss";
@import "Button/Button.pcss";
@import "Checkbox/Checkbox.pcss";
@import "CryptoIcon/CryptoIcon.pcss";
@import "Input/Input.pcss";
@import "Loader/Loader.pcss";
@import "CloseButton/CloseButton.pcss";
@import "PercentageButton/PercentageButton.pcss";
@import "HeaderItem/HeaderItem.pcss";
@import "Alert/Alert.pcss";
@import "AlertIcon/AlertIcon.pcss";
@import "Pagination/Pagination.pcss";
