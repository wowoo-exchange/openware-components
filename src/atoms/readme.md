Very simple building blocks of matter. These components are independent and highly customizable and canâ€™t be broken down further. 
