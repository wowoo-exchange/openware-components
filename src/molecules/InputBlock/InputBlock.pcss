.cr-input-block {
  background-color: var(--input-block-bg);
  border: 1px solid var(--input-block-border-color);
  color: var(--input-block-color);
  display: flex;
  font-size: 12px;
  font-style: normal;
  font-weight: normal;
  justify-content: space-between;
  line-height: normal;
  padding-bottom: 13px;
  padding-top: 17px;
  padding-left: 17px;
  padding-right: 17px;

  &__input {
    border-bottom: 1px solid var(--input-block-input-border-color);

    input {
      background-color: inherit;
      color: var(--input-block-input-color);
    }
  }
}
