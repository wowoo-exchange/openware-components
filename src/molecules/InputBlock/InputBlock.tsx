import classnames from 'classnames';
import * as React from 'react';
import { Input } from '../../atoms';

interface InputBlockProps {
    /**
     * Additional class name for styling. By default element receives `cr-input-block` class
     * @default empty
     */
    className?: string;
    /**
     * Code of the cryptocurrency
     * @default empty
     */
    currency?: string;
    /**
     * Function for getting value of input tag
     * @default empty
     */
    handleChangeValue: (value: string) => void;
    /**
     * Placeholder for Input component
     * @default empty
     */
    placeholder?: string;
    /**
     * Message for input
     */
    message: string;
    /**
     * Type of Input component
     * @default 'text'
     */
    type?: string;
    /**
     * Value of Input component
     * @default ''
     */
    value: string | number;
    /**
     * Function for handling 'Enter' key
     */
    onKeyPress?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
}

/**
 * Input component with ability to render label with cryptocurrency name.
 */
const InputBlock: React.FunctionComponent<InputBlockProps> = (props: InputBlockProps) => {
    const {
        currency, message, type, placeholder,
        className, value, handleChangeValue,
        onKeyPress,
    } = props;
    const cx = classnames('cr-input-block', className);
    return (
        <div className={cx}>
            {currency} {message}
            <Input
                className="cr-input-block__input"
                type={type}
                placeholder={placeholder}
                value={value}
                onChangeValue={handleChangeValue}
                onKeyPress={onKeyPress}
            />
        </div>
    );
};

export {
    InputBlock,
    InputBlockProps,
};
