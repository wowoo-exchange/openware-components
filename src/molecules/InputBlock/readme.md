InputBlock example:

```js
<InputBlock type="text" currency="BCH" message="Withdrawal Address" placeholder="Address" />
```
