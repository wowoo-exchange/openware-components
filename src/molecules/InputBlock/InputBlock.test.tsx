import { mount, shallow } from 'enzyme';
import * as React from 'react';
import { Input } from '../../atoms';
import { InputBlock, InputBlockProps } from './InputBlock';

const defaultProps: InputBlockProps = {
    handleChangeValue: () => {
        return;
    },
    message: 'Message',
    type: 'text',
    value: '0',
    onKeyPress: jest.fn(),
};

const setup = (props: Partial<InputBlockProps> = {}) =>
    shallow(<InputBlock {...{ ...defaultProps, ...props }} />);

describe('InputBlock', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct className', () => {
        const wrapper = setup();
        expect(wrapper.hasClass('cr-input-block')).toBeTruthy();
    });

    it('should pass along supplied className', () => {
        const className = 'some-class';
        const wrapper = setup({ className });
        expect(wrapper.hasClass(className)).toBeTruthy();
    });

    it('should render with correct message', () => {
        const wrapper = setup();
        const { children } = wrapper.props();
        expect(children).toContain('Message');
    });

    it('inner Input component should render with correct type', () => {
        defaultProps.type = 'number';
        const change = defaultProps.handleChangeValue;
        const keyPress = defaultProps.onKeyPress;
        const { type } = mount(
            (
                <Input
                    className="cr-input-block__input"
                    type="number"
                    value=""
                    onChangeValue={change}
                    onKeyPress={keyPress}
                />
            )).props();
        expect(type).toContain(defaultProps.type);
    });
});
