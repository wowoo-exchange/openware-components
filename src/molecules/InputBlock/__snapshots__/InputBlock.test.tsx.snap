// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBlock should render 1`] = `
<div
  className="cr-input-block"
>
   
  Message
  <Input
    className="cr-input-block__input"
    onChangeValue={[Function]}
    onKeyPress={[MockFunction]}
    type="text"
    value="0"
  />
</div>
`;
